pyccel/ast/bind_c.py

```diff
@@ -51,333 +59,445 @@ def __init__(self, *args, original_function, **kwargs):
+        super().__init__(var, **kwargs)
 
-    assert(isinstance(func, FunctionDef))
-
-    args    = list(func.arguments)
-    results = [r.var for r in func.results]
-    body    = func.body.body
-    functions = func.functions
-    _results = []
-    interfaces = func.interfaces
-
-    scope = mod_scope.new_child_scope(func.name)
-
-    # Convert array results to inout arguments
-    for r in results:
-        if r.rank > 0 and r not in args:
-            #wrap the array that is returned from the original function
-            array_body, array_vars = wrap_array(r, scope, False)
-            scope.insert_variable(array_vars[-1])
-            scope.insert_variable(r)
-            body = body + tuple(array_body)
-            array_vars.pop(-1)
-            _results += array_vars
-        elif r.rank == 0:
-            _results += [r]
-
-    body = CodeBlock(body)
-
-    if name is None:
-        name = 'bind_c_{}'.format(func.name).lower()
-
-    # ...
-    results_names = [i.name for i in _results]
-    _args = []
-
-    for i_a, a in enumerate(args):
-        a = a.var
-        if not isinstance(a, (Variable, FunctionAddress)):
-            raise TypeError('Expecting a Variable or FunctionAddress type for {}'.format(a))
-        if not isinstance(a, FunctionAddress) and a.rank > 0:
-            # ...
-            additional_args = []
-            for i in range(a.rank):
-                n_name = 'n{i}_{name}'.format(name=a.name, i=i)
-                n_arg  = Variable('int', n_name)
-
-                additional_args += [n_arg]
-
-            shape_new = tuple(additional_args)
-            # ...
-
-            _args += [FunctionDefArgument(a) for a in additional_args]
-
-            a_new = Variable( a.dtype, a.name,
-                              memory_handling = a.memory_handling,
-                              is_argument = True,
-                              is_optional = a.is_optional,
-                              shape       = shape_new,
-                              rank        = a.rank,
-                              order       = a.order,
-                              precision   = a.precision)
-
-            _args.append(FunctionDefArgument(a_new))
+    @property
+    def original_function_argument_variable(self):
+        """
+        The argument which was passed to the function currently being wrapped.
 
+        The Variable which was passed to the function currently being wrapped
+        in a C-Fortran interface. This variable may have a type which is not
+        compatible with C.
+        """
+        return self._original_arg_var
+
+    @property
+    def sizes(self):
+        """
+        The sizes of the array argument in each dimension.
+
+        A tuple containing the variables which describe the number of
+        elements along each dimension of an array argument. These values
+        must be passed to any C-compatible function taking an array as an
+        argument.
+        """
+        return self._sizes
```

Comment:
Why don't we call this `shape` instead of `sizes`?

Reply:
Fixed acdc9945